test.js

```diff
@@ -0,0 +1,137 @@
+// todo: could be part of csg.js
+// todo: should simplify colinear vertices
+// @return true if both polygons are identical
+function comparePolygons(a, b){
+    // First find one matching vertice
+    // We try to find the first vertice of a inside b
+    // If there is no such vertice, then a != b
+    if (a.vertices.length !== b.vertices.length || a.vertices.length === 0) {
+        return false;
+    }
+    let start = a.vertices[0];
+    let index = b.vertices.findIndex(v => {
+        return v._x === start._x && v._y === start._y && v._z === start._z;
+    });
+    if (index === -1) {
+        return false;
+    }
+    // Rearrange b vertices so that they start with the same vertex as a
+    let vs = b.vertices;
+    if (index !== 0) {
+        vs = b.vertices.slice(index).concat(b.vertices.slice(0, index));
+    }
+    // Compare now vertices one by one
+    for (let i =0; i < a.vertices.length; i++) {
+        if (a.vertices[i]._x !== vs[i]._x ||
+            a.vertices[i]._y !== vs[i]._y ||
+            a.vertices[i]._z !== vs[i]._z){return false;}
+    }
+    return true;
+}
+
+function sameCSG(a, b){
+    return containsCSG(a, b) && containsCSG(b, a);
+}
+
+// a contains b if b polygons are also found in a
+function containsCSG(a, b){
+    return a.toPolygons().map(p => {
+        let found = false;
+        let bp = b.toPolygons();
+        for (let i=0; i<bp.length;i++) {
+            if (comparePolygons(p, bp[i])) {
+                found = true;
+                break;
+            }
+        }
+        return found;
+    }).reduce((a,b) => a && b);
+};
+
+// bit of testing on comparePolygons function
+test('comparePolygons on same single vertex', t => {
+    let a = {vertices:[{_x:0,_y:0,_z:0}]};
+    t.true(comparePolygons(a, a));
+});
+
+test('comparePolygons on different vertices', t => {
+    let a = {vertices:[{_x:0,_y:0,_z:0}]}, b = {vertices:[{_x:1,_y:1,_z:1}]};
+    t.false(comparePolygons(a, b));
+});
+
+test('comparePolygons on same polygon', t => {
+    let a = {vertices:[
+        {_x:0,_y:0,_z:0},
+        {_x:1,_y:1,_z:1},
+        {_x:-1,_y:1,_z:1}
+    ]};
+    t.true(comparePolygons(a, a));
+});
+
+test('comparePolygons on same polygon with different vertice order', t => {
+    let a = {vertices:[
+            {_x:0,_y:0,_z:0},
+            {_x:1,_y:1,_z:1},
+            {_x:-1,_y:1,_z:1}
+        ]},
+        b = {vertices:[
+            {_x:-1,_y:1,_z:1},
+            {_x:0,_y:0,_z:0},
+            {_x:1,_y:1,_z:1}
+        ]};
+    t.true(comparePolygons(a, b));
+});
+
+test('comparePolygons on different polygon with same vertice', t => {
+    let a = {vertices:[
+            {_x:0,_y:0,_z:0},
+            {_x:1,_y:1,_z:1},
+            {_x:-1,_y:1,_z:1}
+        ]},
+        b = {vertices:[
+            {_x:0,_y:0,_z:0},
+            {_x:-1,_y:1,_z:1},
+            {_x:1,_y:1,_z:1}
+        ]};
+    t.false(comparePolygons(a, b));
+});
+
+// Constructive operations
+// a + b = c
+let a = CSG.cube({
```

Comment:
Not entirely sure that creating CSG 'shapes' outside out the scope of tests will work correctly: 
all tests are run in parallel in AVA , so it might lead to weird side effects.

Perhaps just refactor it like this, (bonus: it is also a test helper function)
```javascript
function createOperands(){
const a = CSG.cube({
     center: [0,0,0],
     radius: [1,1,1]
 });
const b = CSG.cube({
     center: [2,0,0],
     radius: [1,1,1]
 });
 const c = CSG.cube({
     center: [1,0,0],
     radius: [2,1,1]
 })
 return {a, b, c}
}
```
then you can just use them in each test :
```javascript 
test('CSG.union', t => {
    const {a, b, c} = createOperands()
     sameCSG(a.union(b), c) ? t.pass() : t.fail();
 });

```